dts/arm/st/f0/stm32f0.dtsi

```diff
@@ -293,6 +293,14 @@
 				#pwm-cells = <2>;
 			};
 		};
+
+		dma1: dma@40020000 {
+			compatible = "st,stm32-dma";
+			reg = <0x40020000 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x1>;
+			status = "disabled";
+			label = "DMA_0";
```

Comment:
I'm curious about the label numeration here. Why not calling fist instance DMA_1?
Usually STM32 devices numeration starts from 1 to fit the reference manuals.

Reply:
I was wanting to name them as DMA_1 and ADC_1 in my first place, while basically there are two reasons:

1. When I was writing applications, I found that CONFIG_DMA_0_NAME and CONFIG_ADC_0_NAME are defined in Kconfig rather than Kconfig.stm32. So DMA_0 is always the first DMA device no matter what driver we are using from kconfig. And the default content of CONFIG_DMA_0_NAME is 'DMA_0'. If we name it as DMA_1, one has to select the **second** DMA device in kconfig, which is DMA_1 to enable the **first** DMA device on STM32, which is DMA1. I think this may bring many bugs in the future.
2. Userspace applications can be shared between devices if all drivers use DMA_0 as its first device (What the OS wants).

Reply:
I just noticed that all DMA are going to be enabled if DMA is selected, but ADC can be selected which to use. I think someday DMA will have this feature, too, to select which DMA device we want to enable.

Reply:
>     1. I found that CONFIG_DMA_0_NAME and CONFIG_ADC_0_NAME are defined in Kconfig rather

It was the same for other devices (UART/SPI/...) and it didn't prevent to set correct name for STM32 instances. Besides, once devices are fully moved to dts, CONFIG_DEV_X_NAME are no more required and deleted (cf serial driver example), so there is no more constraint on numbering.
Using UART_1 in zephyr when user looks for UART_2 in the datasheet is heavily confusing and should be avoided.

Reply:
I agree to number it start from 1, and DMA driver should be like this.
While I think my problem was that we configure which serial to use in Kconfig.stm32, but configure which ADC to use in Kconfig under adc folder. It causes naming problems for ADC driver, though not related to DMA driver, ADC now has to be named from 0.

Reply:
We have to merge the selection of which device to use into device's own kconfig for ADC I think, this is going to solve this problem in my head.